hypershift-operator/controllers/hostedcluster/internal/platform/powervs/powervs.go

```diff
@@ -255,6 +272,35 @@ func (p PowerVS) ReconcileCredentials(ctx context.Context, c client.Client, crea
 	if err != nil {
 		return fmt.Errorf("failed to reconcile control plane operator provider creds: %w", err)
 	}
+
+	// Reconcile the platform provider cluster control manager credentials secret by
+	// resolving  the reference from the HostedCluster and syncing the secret in
+	// the control plane namespace.
+	err = c.Get(ctx, client.ObjectKey{Namespace: hcluster.GetNamespace(), Name: hcluster.Spec.Platform.PowerVS.ClusterControlManagerCreds.Name}, &src)
+	if err != nil {
+		return fmt.Errorf("failed to get control plane operator provider creds %s: %w", hcluster.Spec.Platform.PowerVS.ClusterControlManagerCreds.Name, err)
+	}
+	dest = &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: controlPlaneNamespace,
+			Name:      src.Name,
+		},
+	}
+	_, err = createOrUpdate(ctx, c, dest, func() error {
+		srcData, srcHasData := src.Data["ibmcloud_api_key"]
+		if !srcHasData {
+			return fmt.Errorf("cluster control manager credentials secret %q is missing credentials key", src.Name)
+		}
+		dest.Type = corev1.SecretTypeOpaque
+		if dest.Data == nil {
```

Comment:
It is impossible that this condition is true, because we error out `if !srcHasData`